lib/modules/manager/nuget/util.ts

```diff
@@ -52,13 +52,17 @@ export async function getConfiguredRegistries(
   }
 
   const packageSources = nuGetConfig.childNamed('packageSources');
-  if (!packageSources) {
+  const disabledPackageSources = nuGetConfig.childNamed(
+    'disabledPackageSources',
+  );
+
+  if (!packageSources && !disabledPackageSources) {
```

Comment:
`disabledPackageSources` should be parsed in order to check if one of the default registries (there is only "nuget.org" currently) should be disabled.

I've added two tests to cover this case.

Reply:
What should be Renovate's behavior if the only default registry is disabled _and_ there's no other registries specified? What's nuget's behavior in such cases, e.g. if you run it with no package sources plus nuget disabled?

Reply:
Nuget's behavior in such case is that it will simply fail to restore the nuget packages (as if all the sources were unreachable).

I think Renovate should fail as well, but that is really an edge case. If "nuget.org" is disabled and no other registry is specified, I would expect developers to encounter issues on the CI far before calling Renovate.

But in fact, the current implementation in Renovate is that it will fallback to the default registry ("nuget.org") if no registry is retrieved calling `getConfiguredRegistries`:

lib\modules\manager\nuget\artifacts.ts:34
```
async function createCachedNuGetConfigFile(
...
    (await getConfiguredRegistries(packageFileName)) ?? getDefaultRegistries();
```

Reply:
Yes, Renovate by default will use the default registry for a datasource, i.e. trying nuget.org as you describe. Although it's usually best that we match the way each packager manager works, I don't see much use in attempting to replicate its failure in this case.

Reply:
> I don't see much use in attempting to replicate its failure in this case.

I agree. 👍 

So we're fine as it is? I'll let you resolve this conversation if that's OK for you.

Or you want me to modify the `getConfiguredRegistries` function in order to return the default registry in case there is no other `packageSource`?

Reply:
I'm n such case we should skip returning any registryUrls

Reply:
So the code is correct if the following test passes?

```
  it('reads nuget config file without packageSources and with default registry disabled', async () => {
    fs.findUpLocal.mockReturnValue(
      Promise.resolve<string | null>('NuGet.config'),
    );
    fs.readLocalFile.mockResolvedValueOnce(
      codeBlock`
        <configuration>
          <disabledPackageSources>
                <add key="nuget.org" value="true" />
          </disabledPackageSources>
        </configuration>`,
    );

    const registries = await getConfiguredRegistries('NuGet.config');
    expect(registries?.length).toBe(0);
  });
```